### PR TITLE
Add support for kafka headers

### PR DESCRIPTION
**Pull request by gmcoringa**
Add support for kafka headers with minimal changes.

Note that only the profiles that use `SecorKafkaMessageIterator` will support the headers, at the moment only `kafka-2.0.0` profile supports.

Profile `kafka-1.0.0` still use the legacy iterator: `LegacyKafkaMessageIterator`, which is used by profiles that uses older versions of kafka that doesn't support headers.

### CODE REVIEW

**Comment by HenryCaiHaiying** [comment]
This is a great addition for secor.  Can you also update README.md for the support for kafka message headers?

How are you planning to use message header?  Persist them together with key/value in the sequence file?

**Comment by gmcoringa** [comment]
@HenryCaiHaiying README.md update.

I do not plan to persist the headers. We use them to put information about the message schema, this is done by [Schema Registry](https://github.com/hortonworks/registry) from Hortonworks, which we use to store our avro schemas.

**Comment by HenryCaiHaiying** [comment]
If you don't plan to persist the headers, why do you need to expose headers
in secor?  How do you use it in secor?

**Comment by HenryCaiHaiying** [comment]
@gmcoringa Thanks for the explanation, looks fine to me.

Looks like there is one conflict now on Message.java, can you resolve it?

**Comment by cricket007** [comment]
> Profile kafka-1.0.0 still use the legacy iterator: LegacyKafkaMessageIterator, which is used by profiles that uses older versions of kafka that doesn't support headers.

Not sure the semantics of these, but what would it take to get headers supported in Kafka 0.11 ?

File: src/main/java/com/pinterest/secor/message/MessageHeader.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+public class MessageHeader {
+
+	private final String key;
+	private final byte[] value;
+
+	public MessageHeader(String key, byte[] value) {
+		this.key = key;
+		this.value = value;
+	}
```
Comment (cricket007) [inline src/main/java/com/pinterest/secor/message/MessageHeader.java]: Might help if you added a Copy Constructor from [`RecordHeader`](https://github.com/apache/kafka/blob/0.11.0/clients/src/main/java/org/apache/kafka/common/header/internals/RecordHeader.java)

Then you could have `record.headers().forEach(MesageHeader::new)` in other locations

File: src/main/java/com/pinterest/secor/reader/SecorKafkaMessageIterator.java [hunk 2]
```diff
@@ -40,8 +41,10 @@ public Message next() {
             return null;
         } else {
             ConsumerRecord<byte[], byte[]> consumerRecord = mRecordsBatch.pop();
+            List<MessageHeader> headers = new ArrayList<>();
+            consumerRecord.headers().forEach(header -> headers.add(new MessageHeader(header.key(), header.value())));
```
[the remark this answers is not in the document]
Comment (gmcoringa) [reply src/main/java/com/pinterest/secor/reader/SecorKafkaMessageIterator.java]: `Headers` from kafka client extends `Iterable`, which do not have Streams methods

[the remark this answers is not in the document]
Comment (cricket007) [reply src/main/java/com/pinterest/secor/reader/SecorKafkaMessageIterator.java]: `StreamSupport.stream(iterable.spliterator(), false)`?

**Comment by gmcoringa** [comment]
@HenryCaiHaiying rebase done.

Also in my last commit I've added the headers in `KeyValue` class, so writers will be able to use them.

**Comment by HenryCaiHaiying** [comment]
Thanks for the contribution.

**Pull request merged**